src/client/clientlauncher.cpp

```diff
@@ -401,6 +401,8 @@ bool ClientLauncher::launch_game(std::string &error_message,
 
 	if (menudata.name == "" && !simple_singleplayer_mode) {
 		error_message = gettext("Please choose a name!");
+		if (skip_main_menu) 
```

Comment:
This check can be omitted. See below: The error message is always written to `errorstream` after setting its content.